server/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java

```diff
@@ -1134,4 +1134,54 @@ public void test_update_continues_on_error() {
         execute("update test set a = a + 98");
         assertThat(response).hasRowCount(1);
     }
+
+    @Test
+    public void test_update_fail_fast() {
+        execute("create table test (id int, a int CHECK (a < 100)) clustered by (id) into 1 shards");
+        execute("insert into test (id, a) values (1, 1), (2, 2), (3, 3), (0, 0)");
+        execute("refresh table test");
+
+        try (var session = sqlExecutor.newSession()) {
+            session.sessionSettings().allowFailOnPartialWrites(true);
+            assertSQLError(() -> execute("update test set a = a + 98", session))
+                .hasPGError(INTERNAL_ERROR)
+                .hasHTTPError(BAD_REQUEST, 4000)
+                .hasMessageContaining("Failed CONSTRAINT");
+        }
+
+        execute("refresh table test");
+        execute("select a from test order by a");
+        // 1 + 98 < 100 and it's already written as 99 as it's processed before we failed a check.
+        // However, 0 + 98 < 100 but it's not written as 2 failing a check already caused stoppage.
+        assertThat(response).hasRows("0", "2", "3", "99");
+    }
+
+    @Test
+    public void test_bulk_update_by_id_fail_fast() throws Exception {
+        execute("create table t (id int primary key, a int CHECK (a < 100)) clustered by (id) into 1 shards");
+        execute("insert into t (id, a) values (1, 1), (2, 2), (3, 3)");
+        execute("refresh table t");
+
+        try (var session = sqlExecutor.newSession()) {
+            session.sessionSettings().allowFailOnPartialWrites(true);
+            Object[][] bulkArgs = new Object[][]{
+                new Object[]{2},
+                new Object[]{3}
```

Comment:
Please use a valid row after a invalid one to verify that the valid one isn't written.
```suggestion
                new Object[]{2},
                new Object[]{1}
```